components/backends/trtllm/src/dynamo/trtllm/main.py

```diff
@@ -192,6 +194,17 @@ async def init(runtime: DistributedRuntime, config: Config):
     if config.extra_engine_args != "":
         # TODO: Support extra engine args from json file as well.
         arg_map = update_llm_args_with_extra_options(arg_map, config.extra_engine_args)
+
+    # Apply override_engine_args if provided
+    if config.override_engine_args != "":
```

Comment:
If I understand the changes correctly, precedence is not being handled in the way it should. You would expect the command line args to have precedence over the engine config specified in the yaml file, right? If this is the case, then I would expect we can drop the need for this extra command line argument and just re-order the parsing of the configs such that the command line overwrites any configs which are set by the yaml file.

Also, I don't see this command line flag used in this PR. Is this intentional?